### PR TITLE
Micro benchmarks tests for GraphAnalyze

### PR DESCRIPTION
**Pull request by cristinamanum**
Add micro benchmarks tests for GraphAnalyze. 

## Testing/Validation

Tests Added: Yes

### CODE REVIEW

File: test/MicroBenchmarks/RemoteDependencyWalker/CSVWriter.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace NuGet.MicroBenchmarks
+{
+    public class CSVWriter : IDisposable
+    {
```
Comment (erdembayar) [inline test/MicroBenchmarks/RemoteDependencyWalker/CSVWriter.cs:12]: I don't see how CSVWriter used somewhere. Are you logging events with this CSV writer?

**Comment by loic-sharma** [comment]
Reliable benchmarks are notoriously hard (see [this](https://github.com/dotnet/BenchmarkDotNet#reliability)). How did we evaluate the accuracy of these benchmarks? In particular, it appears that we run these benchmarks in a single process - how do we avoid side effects like garbage collection from affecting results?

Out of curiosity, did we consider using [BenchmarkDotNet](https://github.com/dotnet/BenchmarkDotNet)? This is used by the .NET and ASP.NET Core teams (see [this](https://github.com/dotnet/performance) and [this](https://github.com/aspnet/Benchmarks)) for their own benchmarking. In particular, see the .NET team's write up on why they use BenchmarkDotNet: https://github.com/dotnet/performance/blob/4595ade2796e42ea760f03783ea4a01523866fb1/docs/benchmarkdotnet.md

**Comment by nkolev92** [comment]
Hey @cristinamanum 

Is this still something relevant? 
If not I suggest we close it, and re-open if/when it becomes a priority again.

**Comment by nkolev92** [comment]
🔔 

Is this still important?

**Comment by nkolev92** [comment]
I'm gonna close this for now. 
Please re-open if/when we get back to it. 

It's long stale at this point :) 
We are losing *nothing* by closing the PR :)

**Pull request closed without merging**